src/storage/src/manifest/region.rs

```diff
@@ -13,10 +13,152 @@
 // limitations under the License.
 
 //! Region manifest impl
+use std::any::Any;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use common_telemetry::info;
+use object_store::ObjectStore;
+use store_api::manifest::action::ProtocolAction;
+use store_api::manifest::{
+    Manifest, ManifestLogStorage, ManifestVersion, MetaActionIterator, MIN_VERSION,
+};
+
+use crate::error::{ManifestCheckpointSnafu, Result};
 use crate::manifest::action::*;
+use crate::manifest::checkpoint::Checkpointer;
 use crate::manifest::ManifestImpl;
 
-pub type RegionManifest = ManifestImpl<RegionMetaActionList>;
+pub type RegionManifest = ManifestImpl<RegionSnapshot, RegionMetaActionList>;
+
+#[derive(Debug)]
+pub struct RegionManifestCheckpointer {
+    flushed_manifest_version: AtomicU64,
```

Comment:
nitpick: Add a comment to explain what this field means. Is it the version of the manifest with the last region edit or the version of the latest meta action on object store?